### ISSUE TITLE
Automate cutting releases

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api-provider-vsphere#2051]
The goal of this issue is to automate our release process as far as possible.

Currently the process looks like this:
1. [Maintainer] Push release tag
2. [Maintainer] Create draft release:
  a) Create draft release in GitHub
  b) run `hack/release.sh -l`
  c) run `make release-manifests`
  d) push Docker images
3. post-submit ProwJob builds and pushes images [link](https://github.com/kubernetes/test-infra/blob/e2eaeb8aca492f645454281cc4131dda712d1211/config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml#L38C1-L38C1)
4. [Maintainer] Publish release
  a) finalize release notes
  b) publish release

Suggestion for phase 1 of the implementation.
* Step 2.d) should not be done as the ProwJob pushes the images in 3.
* Step 2 can be entirely automated

Process after phase 1

1. [Maintainer] Push release tag
2. [Maintainer] Wait until GitHub action created release and ProwJob pushed images
3. [Maintainer] Publish release
  a) finalize release notes
  b) publish release

This should help a lot to speed up the process and we don't have to run any builds locally anymore.

In phase 2 I would suggest to change the way we publish images to align to community best practices ("image promotion").

Our ProwJob today pushes directly to the production registry. Instead it should push to the staging repository and then we can explicitly approve the image promotion to production via a PR + reviews.

**Comment by sbueringer** [comment]
PRs for image promotion:
* https://github.com/kubernetes/test-infra/pull/30214

**Comment by sbueringer** [comment]
Implementation should be done now. Remaining tasks are:
* [ ] verify post-submit work as expected
* [ ] verify nightly periodics work as expected
* [ ] create a test release to verify the release workflow (probably a v1.8 beta)

**Comment by sbueringer** [comment]
All done and verified with the beta release today.

I'll follow-up with documentation for the release tasks via: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/2120 and I'll open a new issue soon to document how to consum nightly / post-submit images
